statsmodels/discrete/count_model.py

```diff
@@ -0,0 +1,1062 @@
+        linpred = fitted + exposure + offset
+
+        if which == 'mean':
+            return np.exp(linpred) / (1 - np.exp(-np.exp(linpred)))
+        elif which == 'linear':
+            return linpred
+        elif which == 'prob':
+            if count_prob is not None:
+                counts = np.atleast_2d(count_prob)
+            else:
+                counts = np.atleast_2d(np.arange(0, np.max(self.endog)+1))
+            mu = self.predict(params, exog=exog, exposure=exposure,
+                              offset=offset)[:,None]
+            return self.model_dist.pmf(counts, mu, params[-1],
+                self.model_main.parametrization ,self.trunc)
+        else:
+            raise TypeError(
+                "argument wich == %s not handled" % which)
+
+class GenericCensored(CountModel):
+    __doc__ = """
+    Generic Censored model for count data
+
+    %(params)s
+    %(extra_params)s
+
+    Attributes
+    -----------
+    endog : array
+        A reference to the endogenous response variable
+    exog : array
+        A reference to the exogenous design.
+    """ % {'params' : base._model_params_doc,
+           'extra_params' :
+           """offset : array_like
+        Offset is added to the linear prediction with coefficient equal to 1.
+    exposure : array_like
+        Log(exposure) is added to the linear prediction with coefficient
+        equal to 1.
+
+    """ + base._missing_param_doc}
+
+    def __init__(self, endog, exog, offset=None,
+                       exposure=None, missing='none', **kwargs):
+        self.zero_idx = np.nonzero(endog == 0)[0]
+        self.nonzero_idx = np.nonzero(endog)[0]
+        super(GenericCensored, self).__init__(endog, exog, offset=offset,
+                                                  exposure=exposure,
+                                                  missing=missing, **kwargs)
+
+    def loglike(self, params):
+        """
+        Loglikelihood of Generic Censored model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : float
+            The log-likelihood function of the model evaluated at `params`.
+            See notes.
+
+        Notes
+        --------
+
+        """
+        return np.sum(self.loglikeobs(params))
+
+    def loglikeobs(self, params):
+        """
+        Loglikelihood for observations of Generic Censored model
+
+        Parameters
+        ----------
+        params : array-like
+            The parameters of the model.
+
+        Returns
+        -------
+        loglike : ndarray (nobs,)
+            The log likelihood for each observation of the model evaluated
+            at `params`. See Notes
+
+        Notes
+        --------
+
+        """
+        llf_main = self.model_main.loglikeobs(params)
+        
+        llf = np.concatenate((llf_main[self.zero_idx],
+            np.log(1 - np.exp(llf_main[self.nonzero_idx]))))
```

Comment:
as explanation:
`np.exp(llf_main[self.nonzero_idx]))` is the probability for y=0 given x, because the model_main has endog=zeros